connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -296,22 +307,29 @@ private PluginScanResult scanPluginPath(
         for (Class<? extends T> plugin : plugins) {
             if (PluginUtils.isConcrete(plugin)) {
                 // Temporary workaround until all the plugins are versioned.
-                if (Connector.class.isAssignableFrom(plugin)) {
-                    result.add(
-                            new PluginDesc<>(
-                                    plugin,
-                                    ((Connector) plugin.newInstance()).version(),
-                                    loader
-                            )
-                    );
-                } else {
-                    result.add(new PluginDesc<>(plugin, "undefined", loader));
-                }
+                String version =
+                    Versionable.class.isAssignableFrom(plugin) ? ((Connector) plugin.newInstance()).version() : "undefined";
+                result.add(new PluginDesc<>(plugin, version, loader));
+            } else {
+                log.debug("Skipping {} as it is not concrete implementation", plugin);
             }
         }
         return result;
     }
 
+    private <T> Collection<PluginDesc<T>> getServiceLoaderPluginDesc(
+        Class<T> klass,
+        ClassLoader loader
+    ) throws IllegalAccessException, InstantiationException {
+        Collection<PluginDesc<T>> result = new ArrayList<>();
+        for (T impl : ServiceLoader.load(klass, loader)) {
+            String version =
+                impl instanceof Versionable ? ((Versionable) impl).version() : "undefined";
```

Comment:
This logic is in two places. How about a helper method? If it were named `vesrionFor(Class<T> clazz)` then it could be more easily inlined where it's used.